### DISCUSSION TITLE
Overlay on Map after Keyboard Input

### DISCUSSION THREAD

**Discussion by jodoll** [devemux86/cruiser#322] [Q&A] [answered]
Hey everyone, I have the following problem:

I use [JaxeADV's BarButtons](https://jaxeadv.com/barbuttons/), which are basically a Bluetooth keyboard, for controlling my phone while driving. Controlling the App works well, but after the first button press an overlay appears on the map (but not the buttons) and doesn't disappear. The effect looks just like the Android's ripple effect, when you press a button but don't release it. This doesn't stop the app from working, but it makes the map darker, which isn't great when the sun shines bright.

I'm not entirely sure whether it's an issue with the BarButtons or the App, but I do have the same problem with K1 and until recently with K3, but not with any other apps. However, I'm pretty sure I already tried with a regular Bluetooth keyboard and had the same issue (I'll verify this and report back once I'm home, sometime next week).

So enough writing, here are some Screenshots to clarify (Before/After/Gif of Keypress):

<img title="Before Key Press" src="https://github.com/devemux86/cruiser/assets/7491936/59319cc5-b12a-4917-9952-432a436474d1" width="33%"/>
<img title="Affter Key Press" src="https://github.com/devemux86/cruiser/assets/7491936/820dacb6-d220-4654-9b98-7aea42f9fcf1" width="33%"/>
<img title="Gif showing key press of arrow up" src="https://github.com/devemux86/cruiser/assets/7491936/7ffa256f-48c6-4960-997f-df5c9e8d4644" width="33%"/>

I do find it kind of strange that no one else has this problem, or at least I didn't find anything about it, neither in the Kurviger forum, nor here. Maybe I just didn't use the right search terms, in that case I'd be glad if someone could just point me in the right direction.

Technical Information:
Pixel 4 XL
Android 13
Cruiser 3.0.32
BarButtons FW: Tried with both, v3 (11 May 2023) and v4 (11 May 2023)

Let me know if you need to know something else and thanks in advance!

**Comment by devemux86** [comment]
Hi @jodoll,

Thanks for the detailed report!

I can also reproduce it on Android emulator and I will check what happens.

It seems to be related to new Android versions, because I can't see it on older versions of Android.

**Comment by devemux86** [comment]
@jodoll 

I found what Android changed in new versions.
It will work properly in the next Cruiser update.

**Comment by devemux86** [comment]


**Reply by devemux86** [reply]
Please wait for the next Beta version.

**Reply by jodoll** [reply]
Awesome! And sorry for responding this late, I'll sure give it a try, as soon as possible.

**Comment by devemux86** [comment]
@jodoll 

This should work properly now, can you check the latest:
- https://github.com/devemux86/cruiser/discussions/329

**Reply by jodoll** [reply]
Yesterday, I finally had the chance to try the new version on a test ride, and it works perfectly now. Thanks a lot!

**Reply by devemux86** [reply]
@jodoll 

Great to have this resolved and thanks again for your report!

**Comment by devemux86** [comment, marked as answer]
Fixed in [Cruiser 3.0.34](https://play.google.com/store/apps/details?id=gr.talent.cruiser).
